lib/PHPCfg/AstVisitor/NameResolver.php

```diff
@@ -81,7 +84,7 @@ protected function parseTypeDecl($type) {
         }
         $regex = '(^([a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*\\\\)*[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*$)';
         if (!preg_match($regex, $type)) {
-            throw new \RuntimeException("Unknown type declaration found: $type");
+            return false;
```

Comment:
This may interact badly with the recursive calls to `parseTypeDecl`. Shouldn't we instead return `$type` as in the variable case?